Userland/Libraries/LibWeb/HTML/TraversableNavigable.cpp

```diff
@@ -782,6 +782,11 @@ TraversableNavigable::HistoryStepResult TraversableNavigable::apply_the_history_
     // 20. Set traversable's current session history step to targetStep.
     m_current_session_history_step = target_step;
 
+    // Not in the spec:
+    auto back_enabled = m_current_session_history_step > 0;
+    auto forward_enabled = m_current_session_history_step < static_cast<int>(m_session_history_entries.size()) - 1;
```

Comment:
Just to check, it's not possible for `m_session_history_entries` to be empty here, right? Otherwise this could overflow and `forward_enabled` would errantly be set to true.

Reply:
there is always at least on session history entry that is added during traversable creation https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibWeb/HTML/TraversableNavigable.cpp#L108

although I added VERIFY, so it would be easier to catch the problem in case something went completely wrong 😅